### ISSUE TITLE
500 error when updating rest API in chalice-local

### ISSUE THREAD

**Issue by areeves87** [localstack/localstack#4084]
# Type of request: This is a ...

[x ] bug report
[ ] feature request

# Detailed description

Using https://github.com/localstack/chalice-local. If we deploy the default app we get from chalice-local new-project, resources are created successfully. But if we redeploy the app -- which one would like to do after making changes -- we get a 500 error when trying to update the rest API. 

Also see https://github.com/localstack/chalice-local/issues/6

Error from chalice
```
An error occurred (500) when calling the UpdateRestApi operation (reached max 
 retries: 4): <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">

 <title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The
  server encountered an internal error and was unable to complete your request.
  Either the server is overloaded or there is an error in the application.</p>
```

Error from localstack
```
2021-05-28T04:16:34:INFO:localstack.services.apigateway.apigateway_starter: Missing "value" in JSONPatch operation for {'id': 'tbec4lw1ni', 'region_name': 'us-east-1', 'name': None, 'description': None, 'create_date': 1622175385, 'api_key_source': 'HEADER', 'policy': None, 'endpoint_configuration': {'types': ['EDGE']}, 'tags': {}, 'deployments': {'p8ph035gls': {'id': 'p8ph035gls', 'stageName': 'api', 'description': '', 'createdDate': 1622175385}}, 'authorizers': {}, 'stages': {'api': {'stageName': 'api', 'deploymentId': 'p8ph035gls', 'methodSettings': {}, 'variables': {}, 'description': '', 'cacheClusterEnabled': False}}, 'resources': {'yg8m1ifn11': <moto.apigateway.models.Resource object at 0x7fd3ceb8eeb0>}, 'models': {}}: {'op': 'replace', 'path': '/minimum_compression_size', 'value': ''}
```
localstack info
```
LocalStack version: 0.12.11
LocalStack Docker container id: 82cdea04732a
LocalStack build date: 2021-05-28
LocalStack build git hash: b6c21a69
```

## Expected behavior

chalice-local updates the api endpoints with any changes to app.py

## Actual behavior

500 error from localstack

# Steps to reproduce

## Command used to start LocalStack

`localstack start`

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

```
chalice-local new-project helloworld
cd helloworld
chalice-local deploy # successful
chalice-local deploy # 500 error
```

**Comment by areeves87** [comment]
The following hack gives the desired behavior. 

If I change the api gateway call from chalice, I can successfully update the api with consecutive calls to `chalice-local deploy`. Change chalice.awsclient.TypedAWSClient.update_rest_api to always set the patch_operations to empty list like so:

```python
    def update_rest_api(self, rest_api_id, patch_operations):
        # type: (str, List[Dict]) -> None
        client = self._client('apigateway')
        client.update_rest_api(
            restApiId=rest_api_id,
            # patchOperations=patch_operations
            patchOperations=[]
        )
```

**Comment by whummer** [comment]
Thanks for reporting @areeves87 . This should be fixed in #4151 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks!
